test/support/src/helpers.h

```diff
@@ -759,6 +759,47 @@ std::string get_fragment_dir(std::string array_dir);
  */
 std::string get_commit_dir(std::string array_dir);
 
+/**
+ * Base class that lists the given path, ensures each directory has the
+ * expected file extensions, and stores the counts of each file type.
+ */
+class FileCount {
+ public:
+  // Constructors and destructors.
+  FileCount() = delete;
+  FileCount(
+      VFS vfs, std::string path, std::vector<std::string> expected_extensions);
+  ~FileCount() = default;
+
+  // Private member accessors.
+  const std::map<std::string, uint64_t>& file_count() const;
+  uint64_t file_count(std::string name);
+  uint64_t dir_size();
+
+ private:
+  // Store the number of each file type.
+  std::map<std::string, uint64_t> file_count_;
+  uint64_t dir_size_;
+};
+
+/* Inherit the FileCount base class for commits file extensions. */
+class CommitsDirectory : public FileCount {
+ public:
+  // Constructors and destructors.
+  CommitsDirectory() = delete;
```

Comment:
Instead of taking a path to the commits directory, let's take a path to the array, that way each callers don't have to figure out the commits directory.